src/databases/functions/users/user-block-by-admin.js

```diff
@@ -0,0 +1,7 @@
+const {User} = require('../../models');
+
+module.exports = async (userId) => {
+  if (userId == null) return null;
+  const result = await User.update({blocked_by_admin: true}, {where: {user_id: userId}});
+  return result;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `user-block-by-admin.js` is missing error handling. If the `User.update` operation fails, it will throw an unhandled promise rejection. It's a good practice to wrap async operations in a try-catch block to handle any potential errors.

```diff
const {User} = require('../../models');

module.exports = async (userId) => {
  if (userId == null) return null;
+  try {
    const result = await User.update({blocked_by_admin: true}, {where: {user_id: userId}});
+     return result;
+  } catch (error) {
+     console.error(`Failed to block user with id ${userId}: `, error);
+     throw error;
+  }
-  return result;
};
```
Also, consider using strict equality (`===`) instead of loose equality (`==`) when checking for `null`. Loose equality can lead to unexpected results due to type coercion.


<!-- This is an auto-generated comment by OSS CodeRabbit -->